✨ Basic Race features implementation

orik1504 opened the pull request:
Development branch for #2

orik1504 committed:
✨ Added implementation of `PlayerRaceResult`

RealA10N commented on TypeRaceAir/race.py:
```diff
@@ -0,0 +1,63 @@
+import typing
```
Please add a docstring that describes the module (file)

orik1504 replied:
like a "readme" doc about the module?


> Please add a docstring that describes the module (file)

RealA10N commented on TypeRaceAir/race.py:
```diff
@@ -0,0 +1,63 @@
+import typing
+
+
+class Race():
+
+    def __init__(self, sentece: str, players: typing.List[Player]):
+        self.sentece = sentece
+        self.players = players
+
+
+class Player():
+    pass
```
Finish implementing the `Race` and `Player` objects

RealA10N commented on TypeRaceAir/race.py:
```diff
@@ -0,0 +1,63 @@
+
+
+class PlayerRaceResult():
+    def __init__(self, race: Race, player: Player, submit_timestamp: float, submit_sentence: str):
```
If there are too many arguments in a single line, consider using the following syntax
```suggestion
    def __init__(self,
        race: Race,
        player: Player,
        submit_timestamp: float,
        submit_sentence: str
        ):
```

RealA10N commented on TypeRaceAir/race.py:
```diff
@@ -0,0 +1,63 @@
+    pass
+
+
+class PlayerRaceResult():
```
Please document each object in general. Why does it exist? Where it will be used? What does it do?

RealA10N commented on TypeRaceAir/race.py:
```diff
@@ -0,0 +1,63 @@
+            if cur_submitted == cur_target
+        )
+
+    @ property
```
Please lint your files before check-in
```suggestion
    @property
```

RealA10N commented on TypeRaceAir/race.py:
```diff
@@ -0,0 +1,63 @@
+        total_words = len(self.race.sentece.split())
+        return self.words_correct/total_words
+
+    @ property
```
```suggestion
    @property
```

RealA10N replied:
I'm not fixing all linting problems. Please go over them and fix them yourself

RealA10N commented on TypeRaceAir/race.py:
```diff
@@ -0,0 +1,63 @@
+        return (60/self.time)*self.words_correct
+
+    @ property
+    def score(self,) -> int:
+        '''The total calculated score of the current player in the current race. '''
+        return
```
**???**
If not implemented yet, use the `pass` keyword (or even better - don't check-in)

RealA10N reviewed (changes requested):
Great work! 🥳🥳
Please take a look at my comments

RealA10N commented:
Also, if possible, try adding tests to test the new features you implement.
Add them under the `tests` directory, outside of the main code.
Take a look at the `pytest` module [here](https://docs.pytest.org/en/stable/)

orik1504 committed:
✨started working on player and race classes

started working on player and race classes, fixed linting and finished PlayerRaceResult class

orik1504 committed:
✨change the look of arguments at PlayerRaceResult init

orik1504 committed:
⚡Updated implementation

RealA10N commented on TypeRaceAir/race.py:
```diff
@@ -0,0 +1,88 @@
+# -- IMPORTS --#
```
This comment is unnecessary

RealA10N commented on TypeRaceAir/race.py:
```diff
@@ -0,0 +1,88 @@
+class Race():
+
+    def __init__(self, sentece: str, players: typing.List[Player]):
+        '''The race sentence'''
+        self.sentece = sentece
+        '''	The players that participate in the race.'''
```
Please use regular inline comments inside code blocks. Use docstring only at the begging of a method, function, or object. More about docstrings [here](https://www.python.org/dev/peps/pep-0257/)

RealA10N commented:
@orik1504 I think we might want to change the way that we check the accuracy of the sentence typed by the player.
Please read the following articles to learn about the Levenshtein Distance Equation.
Please also note that this equation assumes by default that the input strings are equal in length (which may not be the case in our program). Try to find a solution to this problem.
Good luck! 😊

- [Levenshtein Distance Equation](https://en.wikipedia.org/wiki/Levenshtein_distance)
- [Understanding the Levenshtein Distance Equation for Begginers](https://medium.com/@ethannam/understanding-the-levenshtein-distance-equation-for-beginners-c4285a5604f0)

RealA10N committed:
✏ Fixed typo, updated linting
